internal/exec/exec.go

```diff
@@ -72,21 +88,29 @@ func (r *Request) execSelections(ctx context.Context, sels []selected.Selection,
 	var fields []*fieldToExec
 	collectFieldsToResolve(sels, s, resolver, &fields, make(map[string]*fieldToExec))
 
+	defer func() {
+		for _, f := range fields {
+			if f.out != nil {
+				BytePool.Put(f.out.Bytes())
+			}
+		}
+	}()
+
 	if async {
 		var wg sync.WaitGroup
 		wg.Add(len(fields))
 		for _, f := range fields {
 			go func(f *fieldToExec) {
```

Comment:
I remember us adding some code here for bounding the number of goroutines. What happened to it?

Reply:
Our change only limited the number of list items we handle concurrently, not fields:
https://github.com/sourcegraph/graphql-go/blob/e67ac72893d245ce10092afba20bd2289a2c31c1/internal/exec/exec.go#L341